enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,285 @@
+### Non-Goals
+
+- Allow cluster admins to customize categories or hide individual items.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add `spec.customization.developerCatalog.types.state` suggests types are `Enabled` or `Disabled`,
+`spec.customization.developerCatalog.types.disabled` which lists types to hide and 
+`spec.customization.developerCatalog.types.enabled` which lists types to show.
+
+### Workflow Description
+
+To enable the types:
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog
+|           ├── types
+|               ├── state
+|               ├── enabled
+|                   └──
+└── ...
+```
+
+To disable the types:
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog
+|           ├── types
+|               ├── state
+|               ├── disabled
+|                   └──
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `types: object` , defines the types to enable or disable.
+
+// +optional
+- `state: Enabled|Disabled` , union discriminator suggests types are `Enabled` or `Disabled`. Default value is `Enabled`.
+
+// +optional
+- `disabled: string[]` , defines a list of types to hide. 
+
+// +optional
+- `enabled: string[]` , defines a list of types to enable. 
+
+If the `state` is `Enabled` and `enabled` list is empty, all the types will be shown.
+If the `state` id `Enabled` and `enabled` list is non-empty, only the `enabled` list items will be shown.
+If the `state` is `Disabled` and `disabled` list is empty, all the types will be hidden.
+If the `state` id `Disabled` and `disabled` list is non-empty, only the `disabled` list items will be hidden.
+
+A type should not be listed both in `enabled` and `disabled` lists.
+
+To disable the complete developer catalog then in `disabled` all the available types
+should be listed or make th `disabled` list empty for `state: Disabled`.
+
+Types will vary based on the operator installed.
+
+### API Extensions
+
+Extend the existing `operator.openshift.io/v1` / `Console CRD` by adding `spec.customization.developerCatalog.types.state` 
+suggests types are `Enabled` or `Disabled`,
+`spec.customization.developerCatalog.types.disabled` which lists types to hide and 
+`spec.customization.developerCatalog.types.enabled` which lists types to show.
+
+So the example yaml should look like this one:
+
+To enable the types:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    developerCatalog:
+      types:
+        state: Disabled
+        disabled: 
+          - Devfile
+          - HelmChart
+  ...
+```
+
+To disable the types:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    developerCatalog:
+      types:
+        state: Enabled
+        enabled: 
+          - Devfile
+          - HelmChart
+  ...
+```
+
```

Comment:
Can we have one more example here for disable everything:


```suggestion
To disable the complete Developer Catalog and all sub-catalogs:

```yaml
apiVersion: operator.openshift.io/v1
kind: Console
metadata: 
  name: cluster
  ...
spec: 
  customization:
    developerCatalog:
      types:
        state: Enabled
        enabled: [] // Enabling no specific type means that all types are disabled!
```
```